app/assets/javascripts/discourse/app/components/user-menu/menu.hbs

```diff
@@ -0,0 +1,50 @@
+<div class="user-menu revamped menu-panel drop-down" data-max-width="320">
+  <div class="panel-body">
+    <div class="panel-body-contents">
+      <div
+        id={{concat "quick-access-" this.currentTabId}}
+        class="quick-access-panel"
+        tabindex="-1"
+        aria-labelledby={{concat "user-menu-button-" this.currentTabId}}>
+        <this.currentPanelComponent/>
+      </div>
+      <div class="menu-tabs-container" role="tablist" aria-label={{i18n "user_menu.sr_menu_tabs"}}>
+        <div class="top-tabs tabs-list">
+          {{#each this.topTabs as |tab|}}
+            <button
+              class={{if (eq tab.id this.currentTabId) "btn btn-flat btn-icon no-text active" "btn btn-flat btn-icon no-text"}}
+              type="button"
+              role="tab"
+              id={{concat "user-menu-button-" tab.id}}
+              tabindex={{if (eq tab.id this.currentTabId) "0" "-1"}}
+              aria-selected={{if (eq tab.id this.currentTabId) "true" "false"}}
+              aria-controls={{concat "quick-access-" tab.id}}
+              data-tab-number={{tab.position}}
+              {{on "click" (fn this.changeTab tab)}}
+              {{!-- template-lint-disable require-context-role --}}
```

Comment:
what is the reason of this? we do have a role here `role="tab"`

Reply:
`ember-template-lint` complains with this:

```
app/assets/javascripts/discourse/app/components/user-menu/menu.hbs
17:14  error  You have an element with the role of "tab" but it is missing the required (immediate) parent element of "[tablist]". Reference: https://www.w3.org/TR/wai-aria-1.1/#tab.  require-context-role
```

The element _is_ contained in an element with `role="tablist"`, but it's not an immediate child and that seems to be what `ember-template-lint` is complaining about. However, [the specs](https://www.w3.org/TR/wai-aria-1.1/#tab) say it doesn't have to be an immediate child. From the specs:

> Authors MUST ensure [elements](https://www.w3.org/TR/wai-aria-1.1/#dfn-element) with [role](https://www.w3.org/TR/wai-aria-1.1/#dfn-role) tab are contained in, or [owned](https://www.w3.org/TR/wai-aria-1.1/#dfn-owned-element) by, an element with the role tablist.

And the definition of "owned" is:

> Owned Element
An 'owned element' is any DOM descendant of the [element](https://www.w3.org/TR/wai-aria-1.1/#dfn-element), any element specified as a child via [aria-owns](https://www.w3.org/TR/wai-aria-1.1/#aria-owns), or any DOM descendant of the owned child.

I couldn't find a way to satisfy the role so I disabled it 🤷